From 59e211d97eb57150714af9973a3dfc0baae86a13 Mon Sep 17 00:00:00 2001
From: Luke Bakken <luke@bakken.io>
Date: Thu, 20 Jan 2022 09:53:51 -0800
Subject: [PATCH 1/2] Support Elixir 1.13

This is the build error prior to these changes:

```
* rabbit_common (/home/bakkenl/development/rabbitmq/rabbitmq-server/deps/rabbit_common)
  could not find an app file at "_build/dev/lib/rabbit_common/ebin/rabbit_common.app". This may happen if the dependency was not yet compiled or the dependency indeed has no app file (then you can pass app: false as option)
** (Mix) Can't continue due to errors on dependencies
```

Telling `mix` to compile `rabbit_common` ensures that the following
links are created:

```
$ ll deps/rabbitmq_cli/_build/dev/lib/rabbit_common/
total 8
drwxr-xr-x  2 bakkenl bakkenl 4096 Jan 20 09:46 .
drwxr-xr-x 10 bakkenl bakkenl 4096 Jan 20 09:46 ..
lrwxrwxrwx  1 bakkenl bakkenl   33 Jan 20 09:46 ebin -> ../../../../../rabbit_common/ebin
lrwxrwxrwx  1 bakkenl bakkenl   36 Jan 20 09:46 include -> ../../../../../rabbit_common/include
```
---
 deps/rabbitmq_cli/mix.exs                              | 6 +-----
 deps/rabbitmq_ct_client_helpers/rabbitmq-components.mk | 2 +-
 deps/rabbitmq_ct_helpers/rabbitmq-components.mk        | 2 +-
 rabbitmq-components.mk                                 | 2 +-
 4 files changed, 4 insertions(+), 8 deletions(-)

Index: rabbitmq-server-3.9.13/deps/rabbitmq_cli/mix.exs
===================================================================
--- rabbitmq-server-3.9.13.orig/deps/rabbitmq_cli/mix.exs
+++ rabbitmq-server-3.9.13/deps/rabbitmq_cli/mix.exs
@@ -11,7 +11,7 @@ defmodule RabbitMQCtl.MixfileBase do
     [
       app: :rabbitmqctl,
       version: "3.8.0-dev",
-      elixir: ">= 1.10.4 and < 1.13.0",
+      elixir: ">= 1.10.4 and < 1.14.0",
       build_embedded: Mix.env == :prod,
       start_permanent: Mix.env == :prod,
       escript: [main_module: RabbitMQCtl,
@@ -149,19 +149,15 @@ defmodule RabbitMQCtl.MixfileBase do
           end
         end
 
-        # We disable compilation for rabbit_common and amqp_client
-        # because Erlang.mk already built them.
         [
           {
             :rabbit_common,
             path: Path.join(deps_dir, "rabbit_common"),
-            compile: false,
             override: true
           },
           {
             :amqp_client,
             path: Path.join(deps_dir, "amqp_client"),
-            compile: false,
             override: true,
             only: :test
           },
Index: rabbitmq-server-3.9.13/rabbitmq-components.mk
===================================================================
--- rabbitmq-server-3.9.13.orig/rabbitmq-components.mk
+++ rabbitmq-server-3.9.13/rabbitmq-components.mk
@@ -118,7 +118,7 @@ dep_looking_glass = git https://github.c
 dep_prometheus = git https://github.com/deadtrickster/prometheus.erl 06425c21a39c1564164f1cc3fe5bdfa8b23b1f78
 dep_ra = hex 2.0.3
 dep_ranch = hex 2.1.0
-dep_recon = hex 2.5.1
+dep_recon = hex 2.5.2
 dep_redbug = hex 2.0.7
 dep_observer_cli = hex 1.7.1
 dep_stdout_formatter = hex 0.2.4
